darwin/torch/dataset.py

```diff
@@ -141,96 +240,190 @@ def __getitem__(self, index: int):
+            tags = [self.classes.index(a["name"]) for a in annotation if "tag" in a]
+            if len(tags) > 1:
+                raise ValueError(
+                    f"Multiple tags defined for this image ({tags}). "
+                    f"This is not valid in a classification dataset."
+                )
+            if len(tags) == 0:
+                raise ValueError(
+                    f"No tags defined for this image ({self.annotations_path[index]})."
+                    f"This is not valid in a classification dataset."
+                )
+        return {"category_id": tags[0]}
+
+    def measure_weights(self, **kwargs):
+        """Computes the class balancing weights (not the frequencies!!) given the train loader
+        Get the weights proportional to the inverse of their class frequencies.
+        The vector sums up to 1
+
+        Returns
+        -------
+        class_weights : ndarray[double]
+            Weight for each class in the train set (one for each class) as a 1D array normalized
+        """
+        # Collect all the labels by iterating over the whole dataset
+        labels = []
+        for i, filename in enumerate(self.images_path):
+            target = self._map_annotation(i)
+            labels.append(target["category_id"])
+        return self._compute_weights(labels)
+
+
+####################################################################################################
 
 
 class InstanceSegmentationDataset(Dataset):
-    def __init__(self, dataset, split: Path, transforms: Optional[List] = None):
+    def __init__(self, root, split: Path, transform: Optional[List] = None):
         """See superclass for documentation"""
-        super().__init__(dataset=dataset, split=split, transforms=transforms)
-        self.classes = [e.strip() for e in open(str(self.dataset.local_path / "lists/classes_masks.txt"))]
-        # Prepend the default transform to convert polygons to instance masks
-        if self.transforms is None:
-            self.transforms = []
-        self.transforms.insert(0, T.ConvertPolysToInstanceMasks())
-        self.transforms = T.Compose(transforms)
+        super().__init__(root=root, split=split, transform=transform)
+        self.classes = [e.strip() for e in open(str(self.root / "lists/classes_polygon.txt"))]
+        self.convert_polygons = T.ConvertPolygonsToInstanceMasks()
 
     def _map_annotation(self, index: int):
-        """See superclass for documentation"""
+        """See superclass for documentation
+
+        Notes
+        -----
+        The return value is a dict with the following fields:
+            image_id : int
+                Index of the image inside the dataset
+            annotations : list[Dict]
+                List of annotations, where each annotation is a dict with:
+                iscrowd : int
+                    Flag to denote where the annotation more than one instance (can be 0 or 1)
```

Comment:
when does this happen?

Reply:
and why isn't it not a boolean?

Reply:
AFAIK never and its because of COCO legacy compatibility. @almazan correct me if I am wrong :)

Reply:
You're correct, `iscrowd` is only there for COCO compatibility. Since this is not a flag that we have in Darwin, I think we can remove it from here.

Reply:
Removed `iscrowd` from the annotations dict.